torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -24,6 +25,17 @@ SizeNode::SizeNode(torch::lazy::Value input, size_t dim)
   upper_bound_ = xla_node->xla_shape(operand(0).index).dimensions(dim_);
 };
 
+int64_t SizeNode::getDynamicValue() const {
+  torch::lazy::NodePtr cloned =
+      torch::lazy::MakeNode<SizeNode>(operands_[0], dim_);
+  std::vector<XLATensorPtr> dummy_size_tensors = {
+      XLATensor::Create(cloned, *GetDefaultDevice(), at::ScalarType::Long)};
+  // TODO: cache the result
+  // TODO: add real dynamic test
```

Comment:
we might consider doing the dynamic test in the caller: `XLASymIntImpl::operator(bool) or operator(int)`.

Reply:
oh, OK. I think I need unit test to test the `getDynamicValue`. And then we can have a test for `XLASymIntImpl::operator(bool)` and `XLASymIntImpl::operator(int)` to see if they correctly calls `getDynamicValue`.